src/Mod/Arch/importIFCHelper.py

```diff
@@ -353,65 +395,88 @@ def getColorFromMaterial(material):
 
 
 def getColorFromStyledItem(styled_item):
+    """Get color from the IfcStyledItem.
+
+    Returns
+    -------
+    float, float, float, int
+        A tuple with the red, green, blue, and transparency values.
+        If the `IfcStyledItem` is a `IfcDraughtingPreDefinedColour`
+        the transparency is set to 0.
+        The first three values range from 0 to 1.0, while the transparency
+        varies from 0 to 100.
+
+    None
+        Return `None` if `styled_item` is not of type `'IfcStyledItem'`
+        or if there is any other problem getting a color.
+    """
+    if not styled_item.is_a("IfcStyledItem"):
+        return None
```

Comment:
Moving this if statement here makes the two following lines ineffective. This same function is used with both IFC4 and IFC2x3 where an intermediary IfcStyledRepresentation is used (there is a lot of that weidness in IFC). So I'd rather keep the `if not styled_item.is_a("IfcStyledItem"):` statement below the `if styled_item.is_a("IfcStyledRepresentation"):`

Reply:
I only tested this with IFC4. Can you test with an IFC2x3 file before I make the change?

This function is called by two different functions, `getColorFromMaterial` which tests for `IfcStyledRepresentation` already; and `buildRelProductColors` which tests for `IfcStyledItem`.

So, I think this function probably doesn't need any `is_a` checks, as it's already pre-checked by the calling functions.

Reply:
Indeed the if statement could also be removed. I'll test with IFC2x3 ASAP